File: RepoTransBench/target_projects/Java/Go/czz1233_GBByPass/public_tests/random_util_full_public_test.go
package public_tests

import (
	"regexp"
	"testing"

	"github.com/stretchr/testify/assert"
	"project/company"
)

func TestRandomString_NegativeLength_Public(t *testing.T) {
	val := company.RandomString(-7)
	assert.Equal(t, "", val, "Negative length should return empty string")
}

func TestRandomString_AllValidCharactersMany_Public(t *testing.T) {
	val := company.RandomString(32)
	assert.NotNil(t, val)
	assert.Equal(t, 32, len(val))
	matched, _ := regexp.MatchString("^[A-Za-z0-9]{32}$", val)
	assert.True(t, matched, "Should be all alphanumeric, 32 chars")
}

File: RepoTransBench/target_projects/Java/C#/czz1233_GBByPass/public_tests/RandomUtilFullPublicTest.cs
using Xunit;
using GBByPass;

namespace PublicTests
{
    public class RandomUtilFullPublicTest
    {
        [Fact]
        public void TestRandomString_negativeLength()
        {
            string val = RandomUtil.RandomString(-7);
            Assert.Equal("", val);
        }

        [Fact]
        public void TestRandomString_allValidCharactersMany()
        {
            string val = RandomUtil.RandomString(32);
            Assert.NotNull(val);
            Assert.Equal(32, val.Length);
            Assert.Matches("^[A-Za-z0-9]{32}$", val);
        }
    }
}